### PR TITLE
Refine app skeleton

### PR DESCRIPTION
**Pull request by KailiDing1121**
- Change a flask-mongo library to get rid of errors
- Remove some typos
- Remove unused imports

### CODE REVIEW

**Comment by iamnwi** [comment]
Looks good! Maybe you can address the pylint issues on `service.py` and `__init__.py` if you have extra time. But I'm good to let it merge into master.

File: service/__init__.py [hunk 1]
```diff
@@ -20,11 +20,10 @@
 """
 import os
 import sys
-import logging
 from flask import Flask
 
 # Get configuration from environment
-DATABASE_URI = os.getenv('DATABASE_URI', 'mongodb://localhost/promotion')
+DATABASE_URI = os.getenv('DATABASE_URI', 'mongodb://localhost:27017')
```
Comment (iamnwi) [inline service/__init__.py]: we can use 'mongodb://localhost' if the Mongo DB is running on the default port 27017

File: service/models.py [hunk 2]
```diff
@@ -60,17 +59,17 @@ class Promotion(db.Document):
...
+    start_date = DateTimeField(required=True)
 
     @classmethod
     def init_db(cls, app):
         """ Initializes the database session """
         cls.logger.info('Initializing database')
         cls.app = app
         # This is where we initialize MongoEngine from the Flask app
-        db.init_app(app)
+        connect('promotion', host='localhost', port=27017)
```
Comment (iamnwi) [inline service/models.py]: we can use `connect('promotion')` if the host is `localhost` and the port is the default port `27017`.

File: service/__init__.py (hunk 1 above)
Comment (KailiDing1121) [reply service/__init__.py]: Thanks! Deleted.

File: service/models.py (hunk 2 above)
Comment (KailiDing1121) [reply service/models.py]: Thanks! Deleted.

**Pull request merged**